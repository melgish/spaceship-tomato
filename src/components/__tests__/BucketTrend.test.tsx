import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import BucketTrend from '../BucketTrend'

const bucket = {
  id: 1,
  name: 'Test Bucket',
  plant: null,
  plantId: 0,
  readings: [],
}

const readings = [
  {
    id: 1,
    bucketId: 1,
    createdAt: new Date(Date.UTC(2024, 5, 2, 9, 48, 25)),
    ph: 6.3,
    ec: 1500,
    tds: 811,
  },
]

describe('When bucket has no readings', () => {
  it('should render', () => {
    const { container } = render(
      <BucketTrend bucket={bucket} xKey="ec" yMin={750} yMax={2000} />,
    )

    expect(container).toMatchSnapshot()
  })
})

describe('When bucket has readings', () => {
  it('should render', () => {
    const { container } = render(
      <BucketTrend bucket={{ ...bucket, readings }} xKey="ec" yMin={750} yMax={2000} />,
    )

    expect(container).toMatchSnapshot()
  })
})
