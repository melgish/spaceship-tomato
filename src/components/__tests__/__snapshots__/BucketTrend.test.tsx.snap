// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`When bucket has no readings > should render 1`] = `
<div>
  <svg
    font-family="sans-serif"
    font-size="10"
    height="156"
    stroke="currentColor"
    text-anchor="middle"
    width="344"
  >
    <g
      transform="translate(32,12)"
    >
      <g
        class="x axis"
        transform="translate(0, 120)"
      >
        <path
          d="M0,6V0H300V6"
          fill="none"
        />
      </g>
      <g
        class="y axis"
        text-anchor="end"
      >
        <path
          d="M-6,120H0V0H-6"
          fill="none"
        />
        <g
          class="tick"
          transform="translate(0, 115.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            800
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 105.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            900
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 96)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1000
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 86.4)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1100
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 76.8)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1200
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 67.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1300
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 57.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1400
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 48)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1500
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 38.4)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1600
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 28.8)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1700
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 19.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1800
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 9.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1900
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 0)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            2000
          </text>
        </g>
      </g>
      <g
        class="data"
        fill="forestgreen"
        stroke="forestgreen"
      >
        <path
          fill="none"
          stroke-width="2"
        />
      </g>
      <g
        class="no-data"
      >
        <text
          stroke="none"
          x="150"
          y="60"
        >
          No Data To Display
        </text>
      </g>
    </g>
  </svg>
</div>
`;

exports[`When bucket has readings > should render 1`] = `
<div>
  <svg
    font-family="sans-serif"
    font-size="10"
    height="156"
    stroke="currentColor"
    text-anchor="middle"
    width="344"
  >
    <g
      transform="translate(32,12)"
    >
      <g
        class="x axis"
        transform="translate(0, 120)"
      >
        <path
          d="M0,6V0H300V6"
          fill="none"
        />
        <g
          class="tick"
          transform="translate(150, 0)"
        >
          <line
            y2="6"
          />
          <text
            dy="0.7em"
            stroke="none"
            y="9"
          >
            Jun 02
          </text>
        </g>
      </g>
      <g
        class="y axis"
        text-anchor="end"
      >
        <path
          d="M-6,120H0V0H-6"
          fill="none"
        />
        <g
          class="tick"
          transform="translate(0, 115.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            800
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 105.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            900
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 96)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1000
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 86.4)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1100
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 76.8)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1200
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 67.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1300
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 57.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1400
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 48)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1500
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 38.4)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1600
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 28.8)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1700
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 19.2)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1800
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 9.6)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            1900
          </text>
        </g>
        <g
          class="tick"
          transform="translate(0, 0)"
        >
          <line
            x2="-6"
          />
          <text
            dy="0.32em"
            stroke="none"
            x="-9"
          >
            2000
          </text>
        </g>
      </g>
      <g
        class="data"
        fill="forestgreen"
        stroke="forestgreen"
      >
        <path
          d="M150,48Z"
          fill="none"
          stroke-width="2"
        />
        <circle
          cx="150"
          cy="48"
          r="4"
        />
      </g>
    </g>
  </svg>
</div>
`;
